STYLE_GUIDE.cs

```diff
@@ -0,0 +1,386 @@
+// | CODING STANDARD-COMPLIANT REFERENCE C# FILE
+// |   _____     ____
+// |  / ___/  __/ / /_
+// | / /__   /_  . __/
+// | \___/  /_  . __/
+// |         /_/_/
+// |
+// |[About this file]
+// |    - This is a 'living document' and will be updated as our standard evolves.
+// |    - For best rendering, be sure to view this file in your text editor rather than through a web site source viewer.
+// |    - The '// |'-style comments denote documentation markup, and are not part of the actual sample.
+// |    - To avoid redundancy, rules inline in the reference code are only mentioned once. Assume they apply generally unless noted, or if obvious from the context.
+// |    - This code is only intended to demonstrate conventions, and as a result sometimes gets nonsensical. Pay no attention to the substance, only the form.
+// |    - Reasoning behind rules are not included here to save space. Ask on Q or browse 'code-conventions + c#' at https://q.unity3d.com/search.html?f=&type=question&redirect=search%2Fsearch&sort=relevance&q=%5Bcode-conventions%5D+and+%5Bc%23%5D.
+// |    - This code will always compile cleanly as a Unity script and will show as fully green in ReSharper using the Unity rule set.
+// |    - This file is currently maintained by scobi and lives in https://ono.unity3d.com/unity-extra/unity-meta/raw/@/ReferenceSource/CSharp/Assets/CSharpReference.cs.
```

Comment:
especially after seeing this line, I think we should just keep the filename identical to the original file: `CSharpReference.cs`

Reply:
I disagree. I renamed it to CODING_STANDARDS.cs. I think that is fair.

This makes it super clear it's not part of source. I see no value in it having the original name.

That line will be good to find where it points to update it.